js/modules/k6/experimental/csv/reader.go

```diff
@@ -0,0 +1,113 @@
+package csv
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"sync/atomic"
+)
+
+// Reader is a CSV reader.
+//
+// It wraps a csv.Reader and provides additional functionality such as the ability to stop reading at a specific line.
+type Reader struct {
+	csv *csv.Reader
+
+	// currentLine tracks the current line number.
+	currentLine atomic.Int64
+
+	// options holds the reader's options.
+	options options
+}
+
+// NewReaderFrom creates a new CSV reader from the provided csv.Reader.
+//
+// It will check whether the first line should be skipped and consume it if necessary.
+// It will also check whether the reader should start from a specific line and skip lines until that line is reached.
+// We perform these operations here to avoid having to do them in the Read method.
+//
+// Hence, this constructor function can return an error if the first line cannot be skipped or if the reader
+// cannot start from the specified line.
+func NewReaderFrom(r io.Reader, options options) (*Reader, error) {
+	if r == nil {
+		return nil, fmt.Errorf("the reader cannot be nil")
+	}
+
+	// Ensure the default delimiter is set.
+	if options.Delimiter == 0 {
+		options.Delimiter = ','
+	}
+
+	csvParser := csv.NewReader(r)
+	csvParser.Comma = options.Delimiter
+
+	reader := &Reader{
+		csv:     csvParser,
+		options: options,
+	}
+
+	var (
+		fromLineSet      = options.FromLine.Valid
+		toLineSet        = options.ToLine.Valid
+		skipFirstLineSet = options.SkipFirstLine
+	)
+
+	// If set, the fromLine option should be greater than 0.
+	if fromLineSet && options.FromLine.Int64 < 0 {
+		return nil, fmt.Errorf("the 'fromLine' option must be greater than 0; got %d", options.FromLine.Int64)
+	}
+
```

Comment:
Should it be "greater" (`>`), or "greater or equal" (`>=`), because the condition doesn't seem to be matching the comment/error message.

Reply:
Good catch indeed, switching to `> 0`, because I think `toLine == 0` doesn't make much sense from a user perspective 🙇🏻

Reply:
I think the logic is almost correct, but we should invert the logic to make the if statement and the error message easier to comprehend without turning our brains to spaghetti indeed.

I'm switching to add:
```
		fromLineIsPositive = fromLineSet && options.FromLine.Int64 > 0
		toLineIsPositive   = toLineSet && options.ToLine.Int64 > 0
```

In the var declaration above, and changing the if statements to `if !fromLineIsPositive` and `if !toLineIsPositive`.